lib/vector-core/src/event/mod.rs

```diff
@@ -39,49 +39,50 @@ pub enum Event {
 }
 
 impl Event {
+    #[must_use]
     pub fn new_empty_log() -> Self {
         Event::Log(LogEvent::default())
     }
 
     pub fn as_log(&self) -> &LogEvent {
         match self {
```

Comment:
Maybe `if let Some() .. else { }` would be better for these ones?

Reply:
Reversed in d949fa1